protocol-specs/utils/chain_state_simulator.js

```diff
@@ -0,0 +1,219 @@
+const {
+	transfer,
+	TransferTransaction,
+	registerDelegate,
+	DelegateTransaction,
+} = require('@liskhq/lisk-transactions');
+const { cloneDeep } = require('lodash');
+const BigNum = require('@liskhq/bignum');
+
+const { createBlock } = require('../utils/blocks');
+const defaultConfig = require('../config/devnet');
+
+class ChainStateSimulator {
+	constructor(
+		genesisBlock,
+		initialAccountsStates,
+		accounts,
+		includeGenesisBlockInState = false,
+	) {
+		this.genesisBlock = genesisBlock;
+		this.previousBlock = genesisBlock;
+		this.state = {
+			chain: includeGenesisBlockInState ? [this.genesisBlock] : [],
+			accounts: cloneDeep(accounts),
+			accountStore: cloneDeep(initialAccountsStates),
+			initialAccountStore: cloneDeep(initialAccountsStates),
+			pendingTransactions: [],
+			appliedTransactions: [],
+		};
+		this.round = 0;
+		this.slot = 0;
+		this.fixedPoint = 10 ** 8;
+	}
+
+	transfer(amount) {
+		return {
+			from: addressFrom => ({
+				to: addressTo => {
+					const amountBedows = `${amount * this.fixedPoint}`;
+					const transferTx = new TransferTransaction(
+						transfer({
```

Comment:
Confusing with the transfer function that surrounds this.
Maybe import `transfer` from `lisk-txs` as `transferLisk`

Reply:
Changed to `transferLisk `